subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -118,6 +120,14 @@ abstract class GeneratePrecompiledScriptPluginAccessors @Inject internal constru
     @get:Input
     abstract val strict: Property<Boolean>
 
+    init {
+        outputs.doNotCacheIf(
+            "Generated accessors can only be cached in strict mode."
```

Comment:
:o: I would make this message more helpful and actionable. We want folks to use the strict mode. A link to a paragraph in https://docs.gradle.org/current/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin would work.

Reply:
I was thinking we should dogfood it first before committing to this path.

Reply:
Sounds good to me. This can be addressed later.